openpype/plugins/publish/collect_scene_version.py

```diff
@@ -26,7 +27,20 @@ class CollectSceneVersion(pyblish.api.ContextPlugin):
         "tvpaint"
     ]
 
+    # in some cases of headless publishing (for example webpublisher using PS)
+    # you want to ignore version from name and let integrate use next version
+    skip_hosts_headless_publish = [
+
+    ]
+
     def process(self, context):
+        # tests should be close to regular publish as possible
+        if (context.data["hostName"] in self.skip_hosts_headless_publish
+                and os.environ.get("HEADLESS_PUBLISH")
+                and not os.environ.get("IS_TEST")):
```

Comment:
Changed order of conditions to first check the main reason and then subconditions of the reason.
```suggestion
        if (
            os.environ.get("HEADLESS_PUBLISH")
            and not os.environ.get("IS_TEST")
            and context.data["hostName"] in self.skip_hosts_headless_publish):
```